### ISSUE TITLE
Invalid PrivateConstructorForUtilityClass warning

### ISSUE THREAD

**Issue by mcalmus** [google/error-prone#530]
I have some (static inner) classes that have no behavior in them, but use an annotation to extend another class that does have behavior. Error-prone is incorrectly flagging these classes with the PrivateConstructorForUtilityClass warning.

With the following code error-prone gives a warning for the BooleanFoo class.

```
public class Foo<E> {
  private E entity;
  public E getEntity() {
    return entity;
  }
  public void setEntity(E anEntity) {
    entity = anEntity;
  }

  public static class BooleanFoo extends Foo<Boolean> implements Serializable {
    private static final long serialVersionUID = 123456789012L;
  }
}
```

**Comment by Stephan202** [comment]
Hit this issue as well. Initially also for a static inner class, but it can be reproduced with top level classes as well. Example:
```java
final class DummyException extends RuntimeException {
    // Without this field the error is not triggered.
    private static final long serialVersionUID = 1L;
}
```

**Comment by mcalmus** [comment]
Confirmed in 2.0.19.
